compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -90,13 +99,59 @@ class ExtractDependencies extends Phase {
             }
         }
         extractDeps.topLevelDependencies.foreach(dep =>
-          recordDependency(sourceFile.file, dep, DependencyContext.DependencyByMemberRef))
+          recordDependency(sourceFile.file, dep._2, DependencyContext.DependencyByMemberRef)(ctx.withOwner(dep._1)))
         extractDeps.topLevelInheritanceDependencies.foreach(dep =>
-          recordDependency(sourceFile.file, dep, DependencyContext.DependencyByInheritance))
+            recordDependency(sourceFile.file, dep._2, DependencyContext.DependencyByInheritance)(ctx.withOwner(dep._1)))
+      }
+    }
+  }
+
+  private def firstClassOrModule(tree: tpd.Tree)(implicit ctx: Context): Option[Symbol] = {
+    import tpd._
+    val acc = new TreeAccumulator[Option[Symbol]] {
+      def apply(x: Option[Symbol], t: Tree)(implicit ctx: Context) =
+        if (x.isDefined) x
+        else t match {
+          case moduleDef: Thicket =>
```

Comment:
I don't think this case is ever going to be hit, objects in Dotty after Typer are represented by a TypeDef and a ValDef, the TypeDef case below will be used.